### PR TITLE
Feature/#330 simplified data output

### LINKED ISSUE #330: Simplification of the data model
_needs better specification_

**This is the the main story for the 2.0 release**

Possibilities:
- leave out an entire table
- leave out an attribute
- aggregate attributes (eg. Shops)
- filter features by attributes
- aggregate geometries
- simplify geometries (generalize, eg streets, administrative areas, ...)
- "Object" conversion (aggragation, eg. Trees -> Forest)

The exported data should be available with simplified data, i.e. leaving out some tables.
The corresponding styles for the QGis need adaptations as well, i.e. the left out tables need to be removed from the style. It is a difference between an empty table and non-extisting data, AFAIK QGis will not be able to load the data in the latter case.

In the simplification, city limits instead of houses and similar should be happening as well.

Depends on #334 

## Ideas

* Create an so-called _pro-mode_: let user choose in various details what to leave out/take in

## Todos

- [ ] enable use of different output *trim levels*
- [ ] implement first (and only?) trim level
    - [ ] leave out table, only include
       - adminarea_a, boundary_l, geoname_l, geoname_p, landuse_a, military_a, military_p, misc_l, natural_a, natural_p, poi_p, pow_p, railway_l, road_l, route_l, utility_p, water_a, water_p, water_l, transport_l, coastline_l, landmass_a, ocean_a
    - [ ] use different sql statements for reduced data
         - [ ] simplify road_l output data: include only:
            - [ ] `track`, `grade1`, `hihgway='track' tracktype='grade1'`
            - [ ] `junction`, `grade2`, `hihgway='track' tracktype='grade2'`
            - [ ] `roundabout`, `motorway`, `hihgway='motorway' junction='roundabout'`
            - [ ] `roundabout`, `motorway_link`, `hihgway='motorway_link' junction='roundabout'`
            - [ ] `roundabout`, `primary`, `hihgway='primary' junction='roundabout'`
            - [ ] `roundabout`, `primary_link`, `hihgway='primary_link' junction='roundabout'`
            - [ ] `unclassified`, `road`, `hihgway='{}' roundabout=*`
            - [ ] `roundabout`, `roundabout`, `hihgway=* junction='roundabout'`
            - [ ] `roundabout`, `secondary`, `hihgway='secondary' junction='roundabout'`
            - [ ] `roundabout`, `secondary_link`, `hihgway='secondary_link' junction='roundabout'`
            - [ ] `roundabout`, `tertiary`, `hihgway='tertiary' junction='roundabout'`
            - [ ] `roundabout`, `trunk`, `hihgway='trunk' junction='roundabout'`
            - [ ] `roundabout`, `trunk_link`, `hihgway='trunk_link' junction='roundabout'`
            - [ ] `roundabout`, `unclassified`, `hihgway='unclassified' junction='roundabout'`
            - [ ] *aggtpyes*: others, highway_links, major_road, minor_road, no_large_vehicule, roundabout, small_road, track

### PR DESCRIPTION
**Pull request by hixi**
Reviewed by:
- [x] @das-g 

Closes #330

### CODE REVIEW

File: osmaxx/conversion/converters/converter_gis/extract/statistics/statistics.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def __init__(self, postgres=None):
         self._stats = []
 
     def gather_all(self):
+        # TODO: react on simplification!
```
Comment (hixi) [inline osmaxx/conversion/converters/converter_gis/extract/statistics/statistics.py]: This TODO awaits input from stakeholders.

File: osmaxx/api_client/conversion_api_client.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def create_parametrization(self, *, boundary, out_format, out_srs):
         Returns:
             A dictionary representing the payload of the service's response
         """
-        json_payload = dict(clipping_area=boundary['id'], out_format=out_format, out_srs=out_srs)
+        json_payload = dict(clipping_area=boundary['id'], out_format=out_format, detail_level=detail_level, out_srs=out_srs)
```
Comment (das-g) [inline osmaxx/api_client/conversion_api_client.py]: Please update the docstring.

File: osmaxx/conversion/converters/converter_gis/sql_tables.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+VIEW_NAMES = [
```
Comment (das-g) [inline osmaxx/conversion/converters/converter_gis/sql_tables.py]: As we also a intermediate view (and might get more of them if we refactor the "filter" scripts), I'd rather name this constant `LAYER_NAMES` or `OUTPUT_LAYER_NAMES`.

Comment (das-g) [inline osmaxx/conversion/converters/converter_gis/sql_tables.py]: Also, why is the module called `sql_tables`, when there's nothing about SQL `TABLE`s in it?

File: tests/conversion/converters/bootstrap_test.py [hunk 4]
```diff
@@ -1,171 +1,103 @@
...
-                    )
-                ) for relative_script_path in expected_script_order
-            ]
-            self.assertListEqual(expected_calls, postgres_mock.execute_sql_file.mock_calls)
+def test_filter_scripts_are_executed_in_correct_order(sql_scripts_filter):
+    bootstrapper = bootstrap.BootStrapper(area_polyfile_string=area_polyfile_string())
+    with mock.patch.object(bootstrapper, '_postgres') as postgres_mock:
+        bootstrapper._filter_data()
+
+        base_path_to_bootstrap = os.path.dirname(bootstrap.__file__)
```
Comment (das-g) [inline tests/conversion/converters/bootstrap_test.py]: `base_path_to_bootstrap` could be extracted to a constant or fixture. Or the `os.path.join(base_path_to_bootstrap, ...)` on the elements returned by the new fixtures could already be performed by those fixtures. I don't see the elements being used without that anywhere.

File: tests/conversion/converters/conftest.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+@pytest.fixture
+def sql_scripts_filter_level_60(sql_scripts_filter):
+    sql_scripts_filter_leveled = sql_scripts_filter
+    replacements = [
+        ('sql/filter/road/010_road.sql', 'sql/filter/road/level-60/010_road.sql')
+    ]
+    for needle, replacement in replacements:
+        sql_scripts_filter_leveled[sql_scripts_filter_leveled.index(needle)] = replacement
+    return sql_scripts_filter_leveled
```
Comment (das-g) [inline tests/conversion/converters/conftest.py]: :+1: nice

File: osmaxx/conversion/converters/converter_gis/sql_tables.py (hunk 3 above)
Comment (hixi) [inline osmaxx/conversion/converters/converter_gis/sql_tables.py]: the answer to the why: the original idea was to store the actual table names that should be included, hence the name. Will rename this to `OUTPUT_LAYER_NAMES` and the module to `layers`.

Comment (das-g) [inline osmaxx/conversion/converters/converter_gis/sql_tables.py]: :+1:

File: osmaxx/api_client/conversion_api_client.py (hunk 2 above)
Comment (hixi) [inline osmaxx/api_client/conversion_api_client.py]: :+1:

**Pull request merged**